dea_check.py

```diff
@@ -231,681 +127,16 @@ def calc_model(model_spec, states, start, stop, T_dea=None, T_dea_times=None):
-    f.close()
-
-    # If run_start is specified this is likely for regression testing
-    # or other debugging.  In this case write out the full predicted and
-    # telemetered dataset as a pickle.
-    if opt.run_start:
-        filename = os.path.join(outdir, 'validation_data.pkl')
-        logger.info('Writing validation data %s' % filename)
-        f = open(filename, 'w')
-        pickle.dump({'pred': pred, 'tlm': tlm}, f, protocol=-1)
-        f.close()
-
-    return plots
-
-
-def plot_cxctime(times, y, fig=None, **kwargs):
-    """Make a date plot where the X-axis values are in CXC time.  If no ``fig``
-    value is supplied then the current figure will be used (and created
-    automatically if needed).  Any additional keyword arguments
-    (e.g. ``fmt='b-'``) are passed through to the ``plot_date()`` function.
-
-    :param times: CXC time values for x-axis (date)
-    :param y: y values
-    :param fig: pyplot figure object (optional)
-    :param **kwargs: keyword args passed through to ``plot_date()``
-
-    :rtype: ticklocs, fig, ax = tick locations, figure, and axes object.
-    """
-    if fig is None:
-        fig = plt.gcf()
-
-    ax = fig.gca()
-    import Ska.Matplotlib
-    ax.plot_date(Ska.Matplotlib.cxctime2plotdate(times), y, **kwargs)
-    ticklocs = Ska.Matplotlib.set_time_ticks(ax)
-    fig.autofmt_xdate()
-
-    return ticklocs, fig, ax
-
-
-def get_power(states):
-    """
-    Determine the power value in each state by finding the entry in calibration
-    power table with the same ``fep_count``, ``vid_board``, and ``clocking``
-    values.
-
-    :param states: input states
-    :rtype: numpy array of power corresponding to states
-    """
-
-    # Make a tuple of values that define a unique power state
-    powstate = lambda x: tuple(x[col] for col in ('fep_count', 'vid_board',
-                                                  'clocking'))
-
-    # dpa_power charactestic is a list of 4-tuples (fep_count vid_board
-    # clocking power_avg).  Build a dict to allow access to power_avg for
-    # available (fep_count vid_board clocking) combos.
-    power_states = dict((row[0:3], row[3])
-                        for row in characteristics.dpa_power)
-    try:
-        powers = [power_states[powstate(x)] for x in states]
-    except KeyError:
-        raise ValueError('Unknown power state: %s' % str(powstate(x)))
-
-    return powers
-
-
-def pointpair(x, y=None):
-    if y is None:
-        y = x
-    return np.array([x, y]).reshape(-1, order='F')
-
-
-def globfile(pathglob):
-    """Return the one file name matching ``pathglob``.  Zero or multiple
-    matches raises an IOError exception."""
-
-    files = glob.glob(pathglob)
-    if len(files) == 0:
-        raise IOError('No files matching %s' % pathglob)
-    elif len(files) > 1:
-        raise IOError('Multiple files matching %s' % pathglob)
-    else:
-        return files[0]
-
-
 if __name__ == '__main__':
     opt, args = get_options()
```

Comment:
I was thinking about that this morning myself--it could certainly be done, though for each tool there may be one or two options that are different, so they'd have to be incorporated somehow (which is doable).